main.js

```diff
@@ -149,12 +149,14 @@ module.exports = function (argumentOptions) {
         connectionOptions.tls = options.tls;
     }
 
-    if (options.routes) {
-        connectionOptions.routes = options.routes;
+    if (options.connections.routes) {
+        console.log("----- received and setting a route config from calling application");
+        server.settings.connections.routes.files = options.connections.routes.files;
     }
 
-    if (options.state) {
-        connectionOptions.state = options.state;
+    if (options.connections.routes.state) {
+        console.log("----- received and setting a state config from calling application");
+        server.settings.connections.routes.state = options.connections.routes.state;
     }
```

Comment:
We typically follow the pattern below. We have these functions defined PAL-Server as an example.

```
/**
 * Checks for the existances of an object.
 *
 * @private
 *
 * @param {*} obj to test
 *
 * @return {boolean} true if obj is not null or undefined
 */
function exists(obj) {
    return obj !== undefined && obj !== null;
}

/**
 * Checks for the existances of a non-null object.
 *
 * @private
 *
 * @param {*} obj to test
 *
 * @return {boolean} true if obj is an object and not null
 */
function existsObject(obj) {
    return typeof obj === 'object' && obj !== null;
}

    /**
     * Checks for the existances of an object and will check if a path
     * exists as part of the object as well.
     *
     * If path is not specified then this is just an alias for exists.
     *
     * @param {*} obj to test
     *
     * @param {string} [path] - The path of the object to look
     * up against the given object. If it is not a string then it is
     * ignored.
     *
     * @return {boolean} true if obj is not null or undefined
     */
    existsPath: function existsPath(obj, path) {
        var current = obj,
            findArraysRegex,
            i,
            len,
            part,
            parts;

        if (typeof path === 'string') {
            findArraysRegex = /\[([0-9]+)\]/g;
            parts = path.replace(findArraysRegex, '.$1').split('.');
            len = parts.length;

            for (i = 0; i < len; i += 1) {
                if (!exists(current)) {
                    break;
                }
                part = parts[i];
                current = current[part];
            }
        }

        return exists(current);
    }
```